### PR TITLE
Added proposal creation

### PR DESCRIPTION
**Pull request by Duddino**
## Abstract

Added  The ability to create proposals inside MPW

## What does this PR address?

You can vote on proposals with MPW, but you couldn't create them. This PR adds the ability to do that

## What features or improvements were added?

Adds the ability to create proposals

## How does this benefit users?

They are no longer required to use the core wallet to create proposals

image
image

### CODE REVIEW

**Review by JSKitty — APPROVED** [review]
**tACK**, incredibly smooth implementation.

Currently waiting for my testnet proposal to pass the `BlockStart` time, other than that, everything appears correct!

The full dataset of my test proposal can be found below:

```json
[
  {
    "Name": "A cool MPW proposal",
    "URL": "https://mypivxwallet.org",
    "Hash": "40fe4759e9c03c66d27a7cc126065e480d9944595a29d1df140d1e5d58956f4d",
    "FeeHash": "a076aac219310e21e99302039376c1cd6bb45dbba20f4a5cacd372950cc22d0b",
    "BlockStart": 1139472,
    "BlockEnd": 1140342,
    "TotalPaymentCount": 6,
    "RemainingPaymentCount": 6,
    "PaymentAddress": "y2QnJZdu5jiAhZWh4CpnSpe5vuXqAneBEW",
    "Ratio": 0,
    "Yeas": 0,
    "Nays": 0,
    "Abstains": 0,
    "TotalPayment": 300,
    "MonthlyPayment": 50,
    "IsEstablished": true,
    "IsValid": true,
    "Allotted": 0
  }
]
```

File: scripts/masternode.js [hunk 1]
```diff
@@ -349,4 +351,165 @@ export default class Masternode {
...
+    static createProposalHash({
+        name,
+        url,
+        nPayments,
+        start,
+        address,
+        monthlyPayment,
+    }) {
+        const end =
+            start + (cChainParams.current.budgetCycleBlocks + 1) * nPayments;
```
Comment (Liquid369) [inline scripts/masternode.js:375]: Probably better with, ```const end = start + ((cChainParams.current.budgetCycleBlocks + 1) * nPayments);

Comment (Liquid369) [reply scripts/masternode.js:375]: Im sorry I need to correct myself its
```const end = start + (cChainParams.current.budgetCycleBlocks * nPayments) + 1;```

**Review by Liquid369 — DISMISSED** [review]
Ignore prior comments, I messed up re-tested and tACK
LGTM!

File: scripts/masternode.js (hunk 1 above)
Comment (Duddino) [reply scripts/masternode.js:375]: Resolved in discord

**Review by panleone — APPROVED** [review]
tACK e52d43de1e4fad59f1f277a257454dc756ec3b45
sorry for the late review

**Pull request merged**